docs/core-concepts.md

```diff
@@ -0,0 +1,52 @@
+# Core Concepts
+
+This document provides a high level overview of the Core Concepts that are embedded in the `cdk-eks-blueprint` framework. For the purposes of this document, we will assume the reader is familiar with Git, Docker, Kubernetes and AWS. 
+
+| Concept       | Description                                                           |     
+|---------------|-----------------------------------------------------------------------|
+| [Blueprint](#blueprint) | A `blueprint` combines `clusters`, `add-ons`, and `teams` into a cohesive object that can deployed as a whole. |                             
+| [Cluster](#cluster) | A Well-Architected EKS Cluster. |
+| [Add-on](#add-on) | A tool or service, represented by Kubernetes or AWS resources, which supports Kubernetes workloads. |
+| [Team](#team) | A logical grouping of IAM identities that have access to a Kubernetes namespace(s). |
+| [Pipeline](#pipeline) | Continuous Delivery pipelines for deploying `clusters` and `add-ons`. |
+| [Application](#application) | An application that runs within an EKS Cluster. |
+
+## Blueprint 
+
+The `cdk-eks-blueprint` framework allows you to configure and deploy what we call `blueprint` clusters. A `blueprint` consists of an EKS cluster, a set of `add-ons` that will be deployed into the cluster, and a set of `teams` who will have access to a cluster. Once a `blueprint` is configured, it can be easily deployed across any number of AWS accounts and regions. 
```

Comment:
It might be reasonable to add that a blueprint also consists of any "workload bootstrapping including installation and maintenance of GitOps tooling and workload onboarding processes"